src/core/public/chrome/nav_links/to_nav_link.ts

```diff
@@ -29,6 +34,28 @@ export function toNavLink(app: PublicAppInfo, basePath: IBasePath): NavLinkWrapp
   });
 }
 
+export function toNavDeepLink(
```

Comment:
Needs a unit test

Reply:
I moved this deepLinks logic into the the `toNavLink` function, I will add some test cases.